pkg/controller/infrastructure/infraflow/ensurer.go

```diff
@@ -426,9 +426,10 @@ func (fctx *FlowContext) ensurePublicIps(ctx context.Context) error {
 	for name, ip := range desiredConfiguration {
 		toReconcile[name] = ip.ToProvider(nameToCurrentIps[name])
 	}
-	for _, inv := range fctx.inventory.ByKind(KindPublicIP) {
-		if ip, ok := nameToCurrentIps[inv]; !ok {
-			fctx.inventory.Delete(*ip.ID)
+	for _, resource := range fctx.inventory.ByKind(KindPublicIP) {
+		if _, ok := nameToCurrentIps[resource.Name]; !ok {
+			log.Info("Removing public IP from inventory", "id", resource)
+			fctx.inventory.Delete(resource.String())
```

Comment:
```suggestion
	for _, id := range fctx.inventory.ByKind(KindPublicIP) {
		if _, ok := nameToCurrentIps[id.Name]; !ok {
			log.Info("Removing public IP from inventory", "id", id.String())
			fctx.inventory.Delete(id.String())
```
Also, looking at the upstream code, `.Name` and `.String()` are not necessarily equal. It's a bit confusing that we have lookups for both, but I guess this is just an adjustment to the new `ByKind()`.

The renaming remark also applies to the other `for _, resource` loops, of course.